File: am/src/main/webapp/js/app/emailtemplate.js
/**
 * 
 */
function EmailTemplate() {
	this.id = null;
	this.name = null;
	this.emailSubject = null;
	this.emailContent = null;
	this.attachments = [];// this is an array of file constants
	// 0: invalid; 1:valid
	this.status = 0;
	this.useCounting = 0;
	this.createDateTime = null;
	this.lastupdate = null;
	this.onlineDateTime = null;
	this.offlineDateTime = null;
	this.parent = null;
	this.owner = null; // organization ID
	this.classtypename = "EmailTemplate";
};

EmailTemplate.prototype = new WorkflowEntity();

EmailTemplate.prototype.parse = function(json) {
	this.id = json.id;
	this.name = json.name;
	this.emailSubject = json.emailSubject;
	this.emailContent = json.emailContent;
	if (json.attachments != null && json.attachments.length > 0) {
		var atts = JSON.parse(json.attachments);
		for (var i = 0; i < atts.length; i++) {
			var fc = new FileConstant();
			fc.parseObject(atts[i]);
			this.attachments.push(fc);
		}
	}
	this.status = json.status;
	this.useCounting = json.useCounting;
	this.createDateTime = Utils.getDateTime(json.createDateTime);
	this.lastupdate = Utils.getDateTime(json.lastupdate);
	this.onlineDateTime = Utils.getDateTime(json.onlineDateTime);
	this.offlineDateTime = Utils.getDateTime(json.offlineDateTime);
	this.parent = json.parent;
	this.owner = json.owner; // organization ID
};

File: core.microservice/src/main/java/com/cloudibpm/core/template/EmailTemplate.java
package com.cloudibpm.core.template;

import com.cloudibpm.core.TreeNode;

public class EmailTemplate extends TreeNode {

	/**
	 * serialVersionUID
	 */
	private static final long serialVersionUID = 1170000652583826091L;
	private String emailSubject = null;
	private String emailContent = null;
	private String attachments = null;
	// 0: invalid; 1:valid
	private int status = 0;
	private int useCounting = 0;
	private long createDateTime = 0L;
	private long lastupdate = 0L;
	private long onlineDateTime = -1;
	private long offlineDateTime = -1;

	public EmailTemplate() {
	}

	public String getEmailSubject() {
		return emailSubject;
	}

	public void setEmailSubject(String emailSubject) {
		this.emailSubject = emailSubject;
	}

	public String getEmailContent() {
		return emailContent;
	}

	public void setEmailContent(String emailContent) {
		this.emailContent = emailContent;
	}

	/**
	 * @return the attachments
	 */
	public String getAttachments() {
		return attachments;
	}

	/**
	 * @param attachments
	 *            the attachments to set
	 */
	public void setAttachments(String attachments) {
		this.attachments = attachments;
	}

	public int getStatus() {
		return status;
	}

	public void setStatus(int status) {
		this.status = status;
	}

	public int getUseCounting() {
		return useCounting;
	}

	public void setUseCounting(int useCounting) {
		this.useCounting = useCounting;
	}

	public long getCreateDateTime() {
		return createDateTime;
	}

	public void setCreateDateTime(long createDateTime) {
		this.createDateTime = createDateTime;
	}

	public long getLastupdate() {
		return lastupdate;
	}

	public void setLastupdate(long lastupdate) {
		this.lastupdate = lastupdate;
	}

	/**
	 * @return the onlineDateTime
	 */
	public long getOnlineDateTime() {
		return onlineDateTime;
	}

	/**
	 * @param onlineDateTime
	 *            the onlineDateTime to set
	 */
	public void setOnlineDateTime(long onlineDateTime) {
		this.onlineDateTime = onlineDateTime;
	}

	/**
	 * @return the offlineDateTime
	 */
	public long getOfflineDateTime() {
		return offlineDateTime;
	}

	/**
	 * @param offlineDateTime
	 *            the offlineDateTime to set
	 */
	public void setOfflineDateTime(long offlineDateTime) {
		this.offlineDateTime = offlineDateTime;
	}

}
